### PR TITLE
buildkite-agent: 2.1.13 -> 2.6.6

### PR DESCRIPTION
**Pull request by zimbatm**
Build the package from source

Re-organise the module a bit to allow custom version of the package

###### Motivation for this change

upgrade

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/continuous-integration/buildkite-agent.nix [hunk 1]
```diff
@@ -20,6 +19,26 @@ in
     services.buildkite-agent = {
       enable = mkEnableOption "buildkite-agent";
 
+      package = mkOption {
+        default = pkgs.buildkite-agent.bin;
```
Comment (dezgeg) [inline nixos/modules/services/continuous-integration/buildkite-agent.nix]: `.bin` isn't needed.

**Comment by domenkozar** [comment]
@zimbatm, nice, are you using this CI? We are planning to soon.

File: nixos/modules/services/continuous-integration/buildkite-agent.nix (hunk 1 above)
Comment (zimbatm) [reply nixos/modules/services/continuous-integration/buildkite-agent.nix]: fixed

**Comment by zimbatm** [comment]
@domenkozar just testing it out for now as I heard good things from a friend. So far I am impressed.

**Review by domenkozar — APPROVED** [review]

**Pull request merged**